Set consistent typecast ENV["VERBOSE"]

bogdanvlviv opened the pull request:
- ~~Refactor DatabaseTasksMigrateTest#test_migrate_receives_correct_env_vars~~
  - ~~Remove redundant overriding ENV["VERBOSE"] in the test.~~

- Refactor AR::Tasks::DatabaseTasks::migrate
  - Set consistent type cast ENV["VERBOSE"]:
      ENV["VERBOSE"] is true if it not equal to "false",
   This also more consistent with the documentation
   http://edgeguides.rubyonrails.org/active_record_migrations.html

- Refactor DatabaseTasksMigrateTest#test_migrate_receives_correct_env_vars
  - Add cases to ensure that environment variables VERBOSE and VERSION have correct typecast.

bogdanvlviv committed:
Refactor AR::Tasks::DatabaseTasks::migrate

Set consistent type cast ENV["VERBOSE"]:
  ENV["VERBOSE"] is true if it not equal "false"

rafaelfranca commented on activerecord/test/cases/tasks/database_tasks_test.rb:
```diff
@@ -339,15 +339,14 @@ def teardown
-      verbose, version = ENV["VERBOSE"], ENV["VERSION"]
+      version = ENV["VERSION"]
 
-      ENV["VERBOSE"] = "false"
```
Why is this redundant?

bogdanvlviv replied:
This test doesn't expect/assert that VERBOSE is `false`. This test assert that `ActiveRecord::Migrator::migrate` is receiving correct attributes inside the method `ActiveRecord::Tasks::DatabaseTasks::migrate`. The attributes related to `VERSION` value but not `VERBOSE`.

rafaelfranca replied:
I see. In this case I think it would be more useful if we change the test to assert VERBOSE too. WDYT?

bogdanvlviv replied:
:+1: I totally agree with that.

bogdanvlviv committed:
Refactor DatabaseTasksMigrateTest#test_migrate_receives_correct_env_vars

Add cases to ensure that environment variables VERBOSE and VERSION have
correct typecast.

bogdanvlviv commented on activerecord/test/cases/tasks/database_tasks_test.rb:
```diff
@@ -343,8 +343,23 @@ def test_migrate_receives_correct_env_vars
+
+      ENV["VERBOSE"] = "yes"
+      ENV["VERSION"] = "unknown"
+      ActiveRecord::Migrator.expects(:migrate).with("custom/path", 0)
```
I am not sure in the current behavior:
When VERSION for example is  "unknown" it becomes `"unknown".to_i # => 0`.
WDYT?

rafaelfranca replied:
I think it is fine. We expect integers.

bogdanvlviv commented:
Thank you!
